### PR TITLE
6. GET /api/reviews/:review_id/comments

### PR DESCRIPTION
**Pull request by dafyddhenke**
Card id: https://trello.com/c/ICJNU717

I went back and forth on whether to use the review model and controller, but I ended up making a separate ones for the comments as I was querying a separate table. That is, the comments table, rather than the review table. 

This includes

- getting an array of the comments by a review id
- they're ordered by the newest comment to the oldest one
- jest test cases 
- correcting a typo from a previous merge

### CODE REVIEW

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
Thinking about the front end, when reading a review or article on a website, we are often shown the comments at the same time. I wouldn't expect two calls to the database at this point. Currently your code is fetching my review, and then fetching the comments by that review id. PSQL is incredibly powerful and will allow us to combine this query. There is nothing wrong here with updating your review model to account for the comments. I'm expecting to see a join when requesting a review by id to show the associated comments.

Refactor your database queries into one appropriate query then please resubmit.

File: __tests__/comments.controller.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  });
+  test("GET /api/reviews/:review_id/comments should output appropriate error messages when passed an invalid id", () => {
+    return request(app)
+      .get("/api/reviews/sfsjhdg/comments")
+      .expect(400)
+      .then((response) => {
+        expect(response.body).toEqual({ msg: "Bad Request" });
+      });
+  });
+  test("GET /api/reviews/:review_id/comments should output appropriate error messages when passed an id that gives no results", () => {
```
[the remark this answers is not in the document]
Comment (dafyddhenke) [reply __tests__/comments.controller.test.js:61]: I don't see 404 as inherently being a problem per se. But if there are no comments, I would say that we have been unable to find any comments. I can change to 204, perhaps.

**Comment by dafyddhenke** [comment]
> Thinking about the front end, when reading a review or article on a website, we are often shown the comments at the same time. I wouldn't expect two calls to the database at this point. Currently your code is fetching my review, and then fetching the comments by that review id. PSQL is incredibly powerful and will allow us to combine this query. There is nothing wrong here with updating your review model to account for the comments. I'm expecting to see a join when requesting a review by id to show the associated comments.

I wouldn't do the front-end that way, not in this context. For one, that would be bad accessibility practice in this context. The better practice would be to show and hide them in a disclosure, with the default being that they were not on screen.

You could cache them after running one query, I suppose, and then dynamically show and hide them. Still, I am not sure that I agree with this feedback. It assumes a design approach that I would not take.

There's otherwise several adjacent tasks with this, so I think one would just end up with some monstrously complex query. 

As otherwise mentioned, I am querying the comments table, not the review table, which is why I have made a separate model. As before, I think it is better to have several separate and slim-line files, rather than one huge one.

**Review by dappernerddesigns — APPROVED** [review]
Nicely done, merge and move on when ready.

**Pull request merged**